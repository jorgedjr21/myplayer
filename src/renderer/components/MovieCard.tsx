import React from 'react';
import { Link } from 'react-router-dom';

interface MovieCardProps {
  id: number;
  title: string;
  image_url?: string;
  rating?: number;
  year: number;
}


const MovieCard: React.FC<MovieCardProps> = ({id, title, image_url, rating, year}) => {
  const defaultImage = "https://via.placeholder.com/400x600?text=No+Image+Available"
  return (
    <Link to={`/movie/${id}`}>
      <div
        className="movie-card relative bg-gray-800 rounded-lg overflow-hidden shadow-lg transform hover:scale-105 transition-transform duration-200 max-w-[400px] xl:max-h-[600px] cursor-pointer"
      >
        {/* Movie Image */}
        <img
          src={image_url || defaultImage}
          alt={title}
          className="w-full h-auto sm:w-[200px] sm:h-[300px] md:w-[240px] md:h-[360px] lg:w-[300px] lg:h-[450px] xl:w-[400px] xl:h-[600px] object-cover"
        />

        {/* Gradient Overlay */}
        <div className="absolute bottom-0 left-0 right-0 h-20 bg-gradient-to-t from-black to-transparent"></div>
        
        {/* Movie Details Overlay */}
        <div className="absolute bottom-4 left-2 right-2 text-white font-semibold text-center px-2 space-y-1">
          <h3 className="text-base sm:text-lg md:text-xl lg:text-2xl xl:text-3xl">{title}</h3>
          <div className="flex items-center justify-center space-x-4 text-sm sm:text-base">
            {/* Year */}
            <span className="bg-gray-900 bg-opacity-75 px-2 py-1 rounded-md">
              {year}
            </span>

            {/* Rating */}
            <span className="flex items-center bg-gray-900 bg-opacity-75 px-2 py-1 rounded-md">
              <svg
                xmlns="http://www.w3.org/2000/svg"
                fill="currentColor"
                viewBox="0 0 24 24"
                className="w-4 h-4 text-yellow-400 mr-1"
              >
                <path d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z" />
              </svg>
              {rating}/10
            </span>
          </div>
        </div>
      </div>
    </Link>
  )
}

export default MovieCard;