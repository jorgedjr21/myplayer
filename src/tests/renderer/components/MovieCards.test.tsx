import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from 'react-router-dom'; // Import MemoryRouter

import "@testing-library/jest-dom"
import MovieCard from '@components/MovieCard'


describe('MovieCard', () => {
  const movieProps = {
    id: 22,
    title: 'Movie Title',
    image_url: 'https://yts.mx/assets/images/movies/vettaiyan_2024/large-cover.jpg',
    rating: 8.8,
    year: 2024
  };

  test('renders correctly with all props', () => {
    render(<MemoryRouter><MovieCard {...movieProps} /></MemoryRouter>);

    const titleElement = screen.getByText(movieProps.title);
    expect(titleElement).toBeInTheDocument();

    const yearElement = screen.getByText(movieProps.year.toString());
    expect(yearElement).toBeInTheDocument();

    const ratingElement = screen.getByText(`${movieProps.rating}/10`);
    expect(ratingElement).toBeInTheDocument();

    const imageElement = screen.getByRole('img', { name: movieProps.title });
    expect(imageElement).toHaveAttribute('src', movieProps.image_url);
  });

  test('matches snapshot', () => {
    const { asFragment } = render(<MemoryRouter><MovieCard {...movieProps} /></MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
  });

  test('displays title, rating, and year', () => {
    render(<MemoryRouter><MovieCard {...movieProps} /></MemoryRouter>);
  
    expect(screen.getByText(movieProps.title)).toBeInTheDocument();
    expect(screen.getByText(`${movieProps.rating}/10`)).toBeInTheDocument();
    expect(screen.getByText(movieProps.year.toString())).toBeInTheDocument();
  });

  test('renders image with correct src and alt attributes', () => {
    render(<MemoryRouter><MovieCard {...movieProps} /></MemoryRouter>);
  
    const imageElement = screen.getByRole('img', { name: movieProps.title });
    expect(imageElement).toHaveAttribute('src', movieProps.image_url);
    expect(imageElement).toHaveAttribute('alt', movieProps.title);
  });

  test('has hover animation class', () => {
    render(<MemoryRouter><MovieCard {...movieProps} /></MemoryRouter>);
    const cardElement = screen.getByText(movieProps.title).parentElement?.parentElement;
    expect(cardElement).toHaveClass('hover:scale-105');
  });

  test('does not display rating if not provided', () => {
    const { rating, ...propsWithoutRating } = movieProps;
    render(<MemoryRouter><MovieCard {...propsWithoutRating} /></MemoryRouter>);
  
    expect(screen.queryByText(`${rating}/10`)).not.toBeInTheDocument();
  });

  test('has accessible image with alt text', () => {
    render(<MemoryRouter><MovieCard {...movieProps} /></MemoryRouter>);
    const imageElement = screen.getByRole('img', { name: movieProps.title });
    expect(imageElement).toHaveAccessibleName(movieProps.title);
  });

  test('renders correctly with a long title', () => {
    const longTitleProps = { ...movieProps, title: 'An Incredibly Long Movie Title That Might Overflow' };
    render(<MemoryRouter><MovieCard {...longTitleProps} /></MemoryRouter>);
    
    expect(screen.getByText(longTitleProps.title)).toBeInTheDocument();
  });

  test('displays default image when image src is not provided', () => {
    const { image_url, ...propsWithoutImageUrl } = movieProps;
    render(<MemoryRouter><MovieCard {...propsWithoutImageUrl} /></MemoryRouter>);
  
    const imageElement = screen.getByRole('img');
    expect(imageElement).toHaveAttribute('src', "https://via.placeholder.com/400x600?text=No+Image+Available"); // Replace with your default image path
  });
  
  
  
});