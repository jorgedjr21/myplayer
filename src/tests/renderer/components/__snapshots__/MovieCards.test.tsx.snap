// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard matches snapshot 1`] = `
<DocumentFragment>
  <a
    href="/movie/22"
  >
    <div
      class="movie-card relative bg-gray-800 rounded-lg overflow-hidden shadow-lg transform hover:scale-105 transition-transform duration-200 max-w-[400px] xl:max-h-[600px] cursor-pointer"
    >
      <img
        alt="Movie Title"
        class="w-full h-auto sm:w-[200px] sm:h-[300px] md:w-[240px] md:h-[360px] lg:w-[300px] lg:h-[450px] xl:w-[400px] xl:h-[600px] object-cover"
        src="https://yts.mx/assets/images/movies/vettaiyan_2024/large-cover.jpg"
      />
      <div
        class="absolute bottom-0 left-0 right-0 h-20 bg-gradient-to-t from-black to-transparent"
      />
      <div
        class="absolute bottom-4 left-2 right-2 text-white font-semibold text-center px-2 space-y-1"
      >
        <h3
          class="text-base sm:text-lg md:text-xl lg:text-2xl xl:text-3xl"
        >
          Movie Title
        </h3>
        <div
          class="flex items-center justify-center space-x-4 text-sm sm:text-base"
        >
          <span
            class="bg-gray-900 bg-opacity-75 px-2 py-1 rounded-md"
          >
            2024
          </span>
          <span
            class="flex items-center bg-gray-900 bg-opacity-75 px-2 py-1 rounded-md"
          >
            <svg
              class="w-4 h-4 text-yellow-400 mr-1"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
            8.8/10
          </span>
        </div>
      </div>
    </div>
  </a>
</DocumentFragment>
`;
